go-controller/pkg/libovsdbops/vswitchd.go

```diff
@@ -0,0 +1,279 @@
+		},
+		ErrNotFound: false,
+		BulkOp:      true,
+	}
+
+	m := newModelClient(vsClient)
+	return m.Delete(opModel)
+}
+
+// CreateOrUpdatePortAndAddToBridge creates or updates the provided Interface
+// Interfae template, provided Port template, and adds the Port to the given bridge
+func CreateOrUpdatePortAndAddToBridge(vsClient libovsdbclient.Client, bridgeUUID string, portTemplate *vswitchdb.Port, ifaceTemplate *vswitchdb.Interface) error {
+	ifaceTemplate.Name = portTemplate.Name
+	bridge := &vswitchdb.Bridge{
+		UUID: bridgeUUID,
+	}
+	opModels := []operationModel{
+		{
+			Model:          ifaceTemplate,
+			OnModelUpdates: onModelUpdatesAllNonDefault(),
+			DoAfter: func() {
+				if ifaceTemplate.UUID != "" {
+					portTemplate.Interfaces = []string{ifaceTemplate.UUID}
+				}
+			},
+			ErrNotFound: false,
+			BulkOp:      false,
+		},
+		{
+			Model:            portTemplate,
+			OnModelMutations: []interface{}{&portTemplate.Interfaces},
+			DoAfter: func() {
+				if portTemplate.UUID != "" {
+					bridge.Ports = []string{portTemplate.UUID}
+				}
+			},
+			ErrNotFound: false,
+			BulkOp:      false,
+		},
+		{
+			Model:            bridge,
+			OnModelMutations: []interface{}{&bridge.Ports},
+			ErrNotFound:      true,
+			BulkOp:           false,
+		},
+	}
+
+	m := newModelClient(vsClient)
+	if _, err := m.CreateOrUpdate(opModels...); err != nil {
+		return fmt.Errorf("failed to create/update ports: %v", err)
+	}
+
+	return nil
+}
+
+// DeleteInterfacesWithPredicate looks up interfaces from the cache based on a
+// given predicate and deletes them
+func DeleteInterfacesWithPredicate(vsClient libovsdbclient.Client, p InterfacePredicate) error {
+	deleted := []*vswitchdb.Interface{}
+	opModel := operationModel{
+		ModelPredicate: p,
+		ExistingResult: &deleted,
+		ErrNotFound:    false,
+		BulkOp:         true,
+	}
+
+	m := newModelClient(vsClient)
+	return m.Delete(opModel)
+}
+
+// FindBridgeByName finds a bridge by name
+func FindBridgeByName(vsClient libovsdbclient.Client, bridgeName string) (*vswitchdb.Bridge, error) {
+	m := newModelClient(vsClient)
+	bridge := &vswitchdb.Bridge{Name: bridgeName}
+	if err := m.Lookup(operationModel{
+		Model:       bridge,
+		ErrNotFound: true,
+	}); err != nil {
+		return nil, err
+	}
+	return bridge, nil
+}
+
+// DeletePorts deletes the given OVS ports by name
+func DeletePort(vsClient libovsdbclient.Client, bridgeName, portName string) error {
+	m := newModelClient(vsClient)
+
+	bridge := &vswitchdb.Bridge{Name: bridgeName}
+	foundPorts := []*vswitchdb.Port{}
+	ops, err := m.LookupOps(nil, operationModel{
+		Model:          &vswitchdb.Port{Name: portName},
+		ExistingResult: &foundPorts,
+		DoAfter: func() {
+			if len(foundPorts) > 0 {
+				bridge.Ports = []string{foundPorts[0].UUID}
+			}
+		},
+		ErrNotFound: false,
+	})
+	if err != nil {
+		return err
+	}
+
+	opModels := []operationModel{
+		{
+			Model:            bridge,
+			OnModelMutations: []interface{}{&bridge.Ports},
```

Comment:
I guess we only need this because the test server has no garbage collection

Reply:
Yep.